compiler/rustc_resolve/src/late/lifetimes.rs

```diff
@@ -375,11 +449,59 @@ impl<'a, 'tcx> Visitor<'tcx> for LifetimeContext<'a, 'tcx> {
         self.labels_in_fn = saved;
     }
 
+    fn visit_fn(
+        &mut self,
+        fk: intravisit::FnKind<'tcx>,
+        fd: &'tcx hir::FnDecl<'tcx>,
+        b: hir::BodyId,
+        s: rustc_span::Span,
+        hir_id: hir::HirId,
+    ) {
+        let name = match fk {
+            intravisit::FnKind::ItemFn(id, _, _, _, _) => id.as_str(),
+            intravisit::FnKind::Method(id, _, _, _) => id.as_str(),
+            intravisit::FnKind::Closure(..) => Symbol::intern("closure").as_str(),
+        };
+        let name: &str = &name;
+        let span = span!(Level::DEBUG, "visit_fn", name);
+        let _enter = span.enter();
+        match fk {
+            // Any `Binders` are handled elsewhere
+            intravisit::FnKind::ItemFn(..) | intravisit::FnKind::Method(..) => {
+                intravisit::walk_fn(self, fk, fd, b, s, hir_id)
+            }
+            intravisit::FnKind::Closure(..) => {
+                self.map.late_bound_vars.insert(hir_id, Vec::default());
+                let scope = Scope::Binder {
```

Comment:
Its interesting that closures didn't get a Binders scope before. When you treat them as fns, they definitely should (just no bound vars).

This also would be needed for some future hypothetical `for<'a> || {}` syntax